scripts/zones/Bastok_Markets/npcs/Teerth.lua

```diff
@@ -2,23 +2,75 @@
 -- Area: Bastok Markets
 --  NPC: Teerth
 --  Guild Merchant NPC: Goldsmithing Guild
--- !zone
 -- !pos -205.190 -7.814 -56.507 235
 -----------------------------------
-local ID = require("scripts/zones/Bastok_Markets/IDs");
+local ID = require("scripts/zones/Bastok_Markets/IDs")
+require("scripts/globals/settings")
+require("scripts/globals/shop")
+require("scripts/globals/status")
 -----------------------------------
 
 function onTrade(player,npc,trade)
-end;
+end
 
 function onTrigger(player,npc)
-    if (player:sendGuild(527,8,23,4)) then
-        player:showText(npc,ID.text.TEERTH_SHOP_DIALOG);
-    end
-end;
+    player:showText(npc,ID.text.TEERTH_SHOP_DIALOG)
+    createShop(player)
+end
 
 function onEventUpdate(player,csid,option)
-end;
+end
 
 function onEventFinish(player,csid,option)
-end;
+end
+
+function createShop(player)
+    local guildRank = player:getSkillRank(dsp.skill.GOLDSMITHING)
+    local stock =
+    {
+        2144,      75,                      0,                          0,      -- Workshop Anvil
+        2143,      75,                      0,                          0,      -- Mandrel
+         642,     200,                      0,                          0,      -- Zinc Ore
+         640,      12,                      0,                          0,      -- Copper Ore
+        1231,      40, dsp.skill.GOLDSMITHING,      dsp.craftRank.RECRUIT,      -- Brass Nugget
```

Comment:
Do we need to include the guild/skill in the table if we're sending the player's guild rank in the call to `dsp.shop.generalGuild(player, stock, guildRank)` down below? `generalGuild` doesn't make use of it.

Alternatively, would the call be better if it were `dsp.shop.generalGuild(player, stock, guild)`, and `generalGuild()` pulls the appropriate player rank when it's called? Then in the individual NPC scripts you'd just need a table of item ID, price, rank, and then call `generalGuild` with the type of shop it is, letting it handle rank stuff for you.
```
local stock =
{
    1231,      40, dsp.craftRank.RECRUIT,  -- Brass Nugget
    661,      300, dsp.craftRank.RECRUIT,  -- Brass Sheet
}
dsp.shop.generalGuild(player, stock, dsp.skill.GOLDSMITHING)
```
Ater which `createShop` wouldn't really be needed, because at that point all the NPC trigger is doing is defining a table and calling a function (like the other NPC shops).

Reply:
Hi @ibm2431 

I'm recreating all the tables again and placing them inside shop.lua as advised yesterday with the following format:

```generalGuildShop = 
{
    [dsp.skill.GOLDSMITHING] =
    {
        [dsp.craftRank.AMATEUR] =
        {
        1231,      40,      -- List of all items available
          661,    300,      -- to amateur
        },
        [dsp.craftRank.RECRUIT] =
        {
        1231,      40,      -- List of all items available
          661,    300,      -- to recruit and below
        }, and so on
```

then they will be called with your suggestion from 23 hours ago:

    ```local guildRank = player:getSkillRank(dsp.skill.GOLDSMITHING)
    local shopItems = dsp.shop.generalGuildShop[dsp.skill.GOLDSMITHING][guildRank]
    dsp.shop.general(player, shopItems)```

Let me know if this is okay so i can continue rewriting it back at home, currently at work for 5 more hours.

Thank you =)

CC @TeoTwawki  in case i'm doing something wrong to not burn more cycles on redoing tables again lol

Reply:
I thought our last discussion we were still using a table field instead of making a table for each rank, and just leaving off the guild ID since that would be the main table ID and was thus redundant. So IBM's example is exactly what I thought we were doing when I left discord chat last night.

Reply:
@kaincenteno 
I thought our last discussion we were still using a table field instead of making a table for each rank, and just leaving off the guild ID since that would be the main table ID and was thus redundant. So IBM's example is almost exactly what I thought we were doing when I left discord chat last night, with the main exception of stuffing the table into shop.lua because you said there were multiple NPC with identical item lists.

```lua
dsp.guildStock =
{
    [dsp.skill.GOLDSMITHING] =
    {
        1231,      40, dsp.craftRank.RECRUIT,  -- Brass Nugget
        661,      300, dsp.craftRank.RECRUIT,  -- Brass Sheet
    }
}
```
```lua
local ID = require("zones\myzone\ids")
require("globals\shop")

function onTrigger(player, npc)
    player:MessageSpecial(ID.wtfEver)
    dsp.shop.generalGuild(player, dsp.guildStock[dsp.skill.GOLDSMITHING])
end
```

Reply:
Changes were made yesterday, please give me your blessing 😸